src/pipeline/pipeline.ml

```diff
@@ -180,10 +180,10 @@ let async_cap_of_prog prog =
      else
        Async_cap.initial_cap()
 
-let infer_prog pkg_opt senv async_cap prog : (Type.typ * Scope.scope) Diag.result =
+let infer_prog ?(viper_mode=false) pkg_opt senv async_cap prog : (Type.typ * Scope.scope) Diag.result =
```

Comment:
I would be nice to get rid of the ?(viper_mode=false) arguments perhaps by adding a field to senv